File: New+Unity+Project_Battery_Texture/New Unity Project/Assets/Scripts/Flashlight and Meters/Flashlight.js
#pragma strict
 static var flashlightOn:boolean = false; 
 var progressTwo: float = 0;
 var lightMeterFull: int = 512;
 var flashlightSoundOn : AudioClip;
 var flashlightSoundOff: AudioClip;
function Start()
{
flashlightOn = true;
Screen.showCursor = false;
}

function Update () 
{
//checks to see how full the flashlight meter is
	progressTwo = flashlightMeter.progressTwo;

//if your flashlight meter is depleted it will not let you turn on the flashlight	
if (progressTwo == lightMeterFull)
{
light.enabled = false;
flashlightOn = false; 
}
else
{
//checks for key input first press turns on the flashlight second press turns it off
if(Input.GetKeyDown("f"))
{
if(light.enabled == true)
{
audio.PlayOneShot(flashlightSoundOff);
light.enabled = false;
flashlightOn = false; 
}
else
{
audio.PlayOneShot(flashlightSoundOn);
flashlightOn = true; 
light.enabled = true;
}
}

}	


}

File: New+Unity+Project_Battery_Texture/New Unity Project/Assets/Scripts/Flashlight/Flashlight.cs
using UnityEngine;
using System.Collections;

public class Flashlight : MonoBehaviour {
	
	public bool on = true;
	
	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
	
		if(Input.GetKeyDown(KeyCode.F)){
        on = !on;
		audio.Play();}
    if(on){
        light.enabled = true;
		
		}
    else if(!on){
        light.enabled = false;
		
		}
	}
}
